amethyst_controls/src/systems.rs

```diff
@@ -211,11 +211,19 @@ impl<'a> System<'a> for MouseFocusUpdateSystem {
 // System which hides the cursor when the window is focused
 pub struct CursorHideSystem {
     event_reader: Option<ReaderId<Event>>,
+    released: bool,
 }
 
 impl CursorHideSystem {
     pub fn new() -> CursorHideSystem {
-        CursorHideSystem { event_reader: None }
+        CursorHideSystem { 
+            event_reader: None,
+            released: false,
+        }
+    }
+
+    pub fn toggle(&mut self) {
```

Comment:
It's pretty rare to have access to the instance of a system. What about having a GrabCursor boolean resource?